torch_xla/experimental/pallas_kernels/quantized_matmul_kernel.py

```diff
@@ -216,6 +205,40 @@ def quantized_matmul_int8(
   # For best performance, only enable this behavior when single input block is used per batch.
   save_q_x = quantize_activation and n_in == 1 and n_out > 1
 
+  acc_dtype = jnp.int32 if quantize_activation else jnp.float32
+
+  # Calculate in/out VMEM size.
+  x_size = batch_block_size * in_block_size * x.dtype.itemsize
+  x_abs_max_val_size = batch_block_size * x_abs_max_val.dtype.itemsize
+  w_size = out_block_size * in_block_size * w.dtype.itemsize
+  scalar_size = out_block_size * scalar.dtype.itemsize
+  out_size = batch_block_size * out_block_size * x.dtype.itemsize
+
+  vmem_in_out = x_size + x_abs_max_val_size + w_size + scalar_size + out_size
+  vmem_in_out *= 2  # Account for compute and vreg spills.
+
+  # Account for double buffering.
+  # Double buffering is used only if there are multiple blocks per in/out.
```

Comment:
is it always guaranteed that double buffering doesn't occurs when the num_blocks==1?

Reply:
Yes. Double buffering means that it's loading the next block while processing current block. If there is only one block, there is no next block to load.